haystack/components/preprocessors/recursive_splitter.py

```diff
@@ -0,0 +1,408 @@
+    >Document(id=..., content: ' Some high-profile applications include advanced web search engines; recommendation systems; interac...', meta: {'original_id': '65167a9823dd883de577e828ca4fd529e6f7241f0ff616acfce454d808478951', 'split_id': 3, 'split_idx_start': 216, '_split_overlap': []})
+    >]
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        *,
+        split_length: int = 200,
+        split_overlap: int = 0,
+        split_unit: Literal["word", "char"] = "word",
+        separators: Optional[List[str]] = None,
+        sentence_splitter_params: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initializes a RecursiveDocumentSplitter.
+
+        :param split_length: The maximum length of each chunk by default in words, but can be in characters.
+            See the `split_units` parameter.
+        :param split_overlap: The number of characters to overlap between consecutive chunks.
+        :param split_unit: The unit of the split_length parameter. It can be either "word" or "char".
+        :param separators: An optional list of separator strings to use for splitting the text. The string
+            separators will be treated as regular expressions unless the separator is "sentence", in that case the
+            text will be split into sentences using a custom sentence tokenizer based on NLTK.
+            See: haystack.components.preprocessors.sentence_tokenizer.SentenceSplitter.
+            If no separators are provided, the default separators ["\n\n", "sentence", "\n", " "] are used.
+        :param sentence_splitter_params: Optional parameters to pass to the sentence tokenizer.
+            See: haystack.components.preprocessors.sentence_tokenizer.SentenceSplitter for more information.
+
+        :raises ValueError: If the overlap is greater than or equal to the chunk size or if the overlap is negative, or
+                            if any separator is not a string.
+        """
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_units = split_unit
+        self.separators = separators if separators else ["\n\n", "sentence", "\n", " "]  # default separators
+        self.sentence_tokenizer_params = sentence_splitter_params
+        self._check_params()
+        self.nltk_tokenizer = None
+        if "sentence" in self.separators:
+            self.warm_up(sentence_splitter_params)
+
+    def warm_up(self, sentence_splitter_params):
+        """
+        Warm up the sentence tokenizer.
+
+        :param sentence_splitter_params: Optional parameters to pass to the sentence tokenizer.
+        :returns:
+            An instance of the SentenceSplitter.
+        """
+        sentence_splitter_params = sentence_splitter_params or {"keep_white_spaces": True}
+        self.nltk_tokenizer = self._get_custom_sentence_tokenizer(sentence_splitter_params)
```

Comment:
And here I'd remove the sentence_splitter_params as a input variable, and the returns docs string since nothing is returned by this method
```suggestion
    def warm_up(self):
        """
        Warm up the sentence tokenizer.
        """
        self.nltk_tokenizer = self._get_custom_sentence_tokenizer(sentence_splitter_params)
```

Reply:
Sorry the failing test is from me forgetting to call `self` on `sentence_splitter_params`

Reply:
no, it's from something else, that one I spotted - it's something in the logic